modules/oe_editorial_corporate_workflow/src/Services/ShortcutTransitionHandler.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_editorial_corporate_workflow\Services;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\content_moderation\ModerationInformationInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\RevisionableInterface;
+use Drupal\Core\Entity\RevisionLogInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\workflows\WorkflowTypeInterface;
+
+/**
+ * Handler for creating state transition revisions when shortcuts are used.
+ */
+class ShortcutTransitionHandler implements ShortcutTransitionHandlerInterface {
+
+  /**
+   * The moderation information service.
+   *
+   * @var \Drupal\content_moderation\ModerationInformationInterface
+   */
+  protected $moderationInfo;
+
+  /**
+   * The current user logged in.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * Entity type manager to get entity storage.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The time service.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
+  /**
+   * Constructs a new StateTransitionValidation.
+   *
+   * @param \Drupal\content_moderation\ModerationInformationInterface $moderation_info
+   *   The moderation information service.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The current user logged in.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity type manager to retrieve storage.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   Time object.
+   */
+  public function __construct(ModerationInformationInterface $moderation_info, AccountProxyInterface $current_user, EntityTypeManagerInterface $entity_type_manager, TimeInterface $time) {
+    $this->moderationInfo = $moderation_info;
+    $this->currentUser = $current_user;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->time = $time;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function handleShortcutTransitions(array $form, FormStateInterface $form_state): void {
```

Comment:
Why are we tying this to a form? The implementation you have simply needs an entity (whatever kind of content entity) and a state to make the revisions. So that's what should be the interface.

Reply:
I need the form state.